lib/rubygems/platform.rb

```diff
@@ -71,7 +71,7 @@ def initialize(arch)
     when String then
       arch = arch.split '-'
 
-      if arch.length > 2 and arch.last !~ /\d/ # reassemble x86-linux-gnu
+      if arch.length > 2 and arch.last !~ /\d+(\.\d+)?$/ # reassemble x86-linux-{libc}
```

Comment:
This makes the code consistent with the line below in terms of what a "version" is:

```
if arch.length == 2 and arch.last =~ /^\d+(\.\d+)?$/ # for command-line
```

I did not pursue further change to limit the scope, but it shows possible room for future enhancements.